llama_index/indices/multi_modal/retriever.py

```diff
@@ -0,0 +1,117 @@
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.vector_store.retrievers.retriever import VectorIndexRetriever
+from llama_index.schema import NodeWithScore
+from llama_index.vector_stores.types import (
+    MetadataFilters,
+    VectorStoreQueryMode,
+)
+
+
+class MutliModalVectorIndexRetriever(VectorIndexRetriever):
+    """Multi Modal Vector index retriever.
+
+    Args:
+        index (MutliModalVectorIndexRetriever): Multi Modal vector store index for images and texts.
+        similarity_top_k (int): number of top k results to return.
+        vector_store_query_mode (str): vector store query mode
+            See reference for VectorStoreQueryMode for full list of supported modes.
+        filters (Optional[MetadataFilters]): metadata filters, defaults to None
+        alpha (float): weight for sparse/dense retrieval, only used for
+            hybrid query mode.
+        doc_ids (Optional[List[str]]): list of documents to constrain search.
+        vector_store_kwargs (dict): Additional vector store specific kwargs to pass
+            through to the vector store at query time.
+
+    """
+
+    def __init__(
+        self,
+        index: MultiModalVectorStoreIndex,
+        similarity_top_k: int = DEFAULT_SIMILARITY_TOP_K,
+        vector_store_query_mode: VectorStoreQueryMode = VectorStoreQueryMode.DEFAULT,
+        filters: Optional[MetadataFilters] = None,
+        alpha: Optional[float] = None,
+        node_ids: Optional[List[str]] = None,
+        doc_ids: Optional[List[str]] = None,
+        sparse_top_k: Optional[int] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        self._index = index
+        self._vector_store = self._index.vector_store
+        # separate image vector store for image retrieval
+        self._image_vector_store = self._index.image_vector_store
+        self._image_embed_model = self._index.image_embed_model
+        self._service_context = self._index.service_context
+        self._docstore = self._index.docstore
+
+        self._similarity_top_k = similarity_top_k
+        self._vector_store_query_mode = VectorStoreQueryMode(vector_store_query_mode)
+        self._alpha = alpha
+        self._node_ids = node_ids
+        self._doc_ids = doc_ids
+        self._filters = filters
+        self._sparse_top_k = sparse_top_k
+
+        self._kwargs: Dict[str, Any] = kwargs.get("vector_store_kwargs", {})
+
+    @property
+    def similarity_top_k(self) -> int:
+        """Return similarity top k."""
+        return self._similarity_top_k
+
+    @similarity_top_k.setter
+    def similarity_top_k(self, similarity_top_k: int) -> None:
+        """Set similarity top k."""
+        self._similarity_top_k = similarity_top_k
+
+    def _retrieve(
+        self,
+        query_bundle: QueryBundle,
+    ) -> List[NodeWithScore]:
+        res = self._text_retrieve(query_bundle)
+        res.extend(self._image_retrieve(query_bundle))
```

Comment:
So we will always retrieve the top_k from text and images

We might need fancier modes in the future :) Reciprocal rerank, simple re-ordering by similarity, etc.

Reply:
oh ya, should we have a separate top_k param for images to start at least?

Reply:
yea that will be the easiest

Reply:
That kind of spirals into having different settings for each retriever though 😓

Reply:
I still kind of think this retriever makes a little more sense as a wrapper around two `VectorStoreIndexRetirever` objects, since we basically are duplicating every function for image vs. text anyways